java/clients/src/main/java/sleeper/clients/admin/deploy/SyncJars.java

```diff
@@ -21,38 +21,43 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import sleeper.util.ClientUtils;
-
 import java.io.IOException;
+import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.Collection;
 import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Objects.requireNonNull;
+import static org.apache.commons.lang3.ObjectUtils.requireNonEmpty;
 
 public class SyncJars {
     private static final Logger LOGGER = LoggerFactory.getLogger(SyncJars.class);
     private final AmazonS3 s3;
     private final Path jarsDirectory;
     private final String bucketName;
-    private final boolean newInstance;
 
     private SyncJars(Builder builder) {
-        s3 = builder.s3;
-        jarsDirectory = builder.jarsDirectory;
-        bucketName = builder.bucketName;
-        newInstance = builder.newInstance;
+        s3 = requireNonNull(builder.s3, "s3 must not be null");
+        jarsDirectory = requireNonNull(builder.jarsDirectory, "jarsDirectory must not be null");
+        bucketName = requireNonEmpty(builder.bucketName, "");
```

Comment:
This could do with an exception message like the variables above

Reply:
Done, thanks